01week/rockPaperScissors.js

```diff
@@ -9,11 +9,59 @@ const rl = readline.createInterface({
 
 
 function rockPaperScissors(hand1, hand2) {
+	const rockPaperScissorsArray = ['rock', 'paper', 'scissors'];
 
-  // Write code here
+	// get rid of whitespace and convert to lower case before beginning to check
+	hand1 = hand1.trim().toLowerCase();
+	hand2 = hand2.trim().toLowerCase();
 
-}
+	// only accept rock, paper, or scissors
+	if (rockPaperScissorsArray.indexOf(hand1) < 0 || rockPaperScissorsArray.indexOf(hand2) < 0) {
+		return "Please choose either rock, paper, or scissors.";
+		getPrompt();
+	} else {
+		if (hand1 === hand2) {
+			return "It's a tie!";
+		}
+
+
+		// if hand1 is rock
+		// and hand2 is scissors, hand1 wins
+		// and hand2 is paper, hand2 wins
+		// otherwise its a tie
+		if (hand1 === 'rock') {
```

Comment:
Best practice would probably to make every if after the first if into an else if. By having them in separate if trees, it's implied that more than one of them could run at a time. Of course here since you're returning in each, only one will run at a time, but it's a good habit to get in to.